tests/end-to-end/regression/2155/Issue2155Test.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\TestFixture\Issue2155;
+
+use function error_log;
+use PHPUnit\Framework\TestCase;
+
+class Foo
+{
+    public function doFoo()
+    {
+        error_log('logged a side effect');
```

Comment:
with the current implementation a test-case which itself fiddles with `ini_set('error_log', ...); ` would render the in this PR added expectation handling useless. I am not sure we can make this more robust, or whether thats good enough.

Reply:
I do not have time at the moment to look into this, especially as I consider this a rather niche use case. I will look into this, though. Thank you for your work so far!

Reply:
Btw: the main case for such new api is, that testing code which calls error_log() currently is considered risky.

The main goal is not asserting  expectations on the error_log()'ed string.

Maybe something like `expectsErrorLog()` or similar would be a even simpler api

Reply:
Sounds reasonable.